bundler/lib/bundler.rb

```diff
@@ -165,6 +169,22 @@ def setup(*groups)
       end
     end
 
+    def install
+      Bundler.definition.specs
+    rescue GemNotFound
+      Bundler.ui.info "Automatically installing missing gems."
```

Comment:
While testing, we were surprised to not see this output when using `require 'bundler/setup'`, but not when calling `bundle exec` or other subcommand that support `auto_install`.

Turns out `require 'bundler/setup'` explicitly does a `Bundler.ui.silence` to silence everything.

I think ideally, this should be shown.

Reply:
Yeah! I was actually surprised by this in the past too, particularly when expecting to see "Resolving gems..." output during `require "bundler/setup"` and it was not there.

If I recall correctly, we couldn't figure out the best path forward, because while it does make sense to me that `require "bundler/setup"` is more silent and does not show messages like "Resolving gems...", I agree we would not want to hide output indicating that it's actually installing gems.

Should we only silence `bundler/setup` output only up to a certain log level?

Reply:
That is an option, but at the same time, I'm not sure what level different log messages are at, or if there would be a level to represent what we want shown 🤔 

I do have this comment calling out that it would normally be silenced:

```
  # this happens in Bundler.setup too, but we call it here early
  # so it can happen without being Bundler.ui.silence'd
```

Reply:
Oh, sorry! So you ended up resolving this in the end. Then this is fine, and we could iterate on a better solution and the comment you left there will be useful for that 👍. Should we add an assertion to the test that checks that the message is printed?

Reply:
Maybe could do something like... remove the `rescue` logic in the method that does the installation, and hoist it to call sites. Or make it a conditional that can be checked

```
if auto_install_needed?
  install
end


def auto_install_needed?
  return false unless settings[:auto_install]
  begin
    definition.specs
    true
  rescue GemNotFound, GitError
     false
  end
end
```

Reply:
That seems fine to me, but at this point would be just cleaner code but would not avoid having to do this both before and during `Bundler.setup`?

Reply:
As minor additional suggestion (besides checking that the log message is printed in the spec), you could remove `Bundler.` prefixes since this now lives inside the `Bundler` module.